packages/headless/src/features/standalone-search-box-set/standalone-search-box-set-actions-loader.ts

```diff
@@ -0,0 +1,90 @@
+import {AsyncThunkAction, PayloadAction} from '@reduxjs/toolkit';
+import {AsyncThunkSearchOptions} from '../../api/search/search-api-client';
+import {standaloneSearchBoxSet} from '../../app/reducers';
+import {SearchEngine} from '../../app/search-engine/search-engine';
+import {
+  registerStandaloneSearchBox,
+  RegisterStandaloneSearchBoxActionCreatorPayload,
+  fetchRedirectUrl,
+  FetchRedirectUrlActionCreatorPayload,
+  updateAnalyticsToSearchFromLink,
+  UpdateAnalyticsToSearchFromLinkActionCreatorPayload,
+  updateAnalyticsToOmniboxFromLink,
+  UpdateAnalyticsToOmniboxFromLinkActionCreatorPayload,
+  StateNeededForRedirect,
+} from './standalone-search-box-set-actions';
+
+export {
+  RegisterStandaloneSearchBoxActionCreatorPayload,
+  FetchRedirectUrlActionCreatorPayload,
+  UpdateAnalyticsToSearchFromLinkActionCreatorPayload,
+  UpdateAnalyticsToOmniboxFromLinkActionCreatorPayload,
+};
+
+/**
+ * The standalone search box set action creators.
+ */
+export interface StandaloneSearchBoxSetActionCreators {
+  /**
+   * Registers a standalone search box.
+   *
+   * @param payload - The action creator payload.
+   * @returns A dispatchable action.
+   */
+  registerStandaloneSearchBox(
+    payload: RegisterStandaloneSearchBoxActionCreatorPayload
+  ): PayloadAction<RegisterStandaloneSearchBoxActionCreatorPayload>;
+
+  /**
+   * Preprocesses the query for the current headless state, and retrieves a redirection URL if a redirect trigger was fired in the query pipeline.
+   *
+   * @param payload - The action creator payload.
+   * @returns A dispatchable action.
+   */
+  fetchRedirectUrl(
+    payload: FetchRedirectUrlActionCreatorPayload
+  ): AsyncThunkAction<
+    string,
+    FetchRedirectUrlActionCreatorPayload,
+    AsyncThunkSearchOptions<StateNeededForRedirect>
+  >;
+
+  /**
+   * Updates the standalone search box analytics data to reflect a search submitted using the search box.
+   *
+   * @param payload - The action creator payload.
+   * @returns A dispatchable action.
+   */
+  updateAnalyticsToSearchFromLink(
+    payload: UpdateAnalyticsToSearchFromLinkActionCreatorPayload
+  ): PayloadAction<UpdateAnalyticsToSearchFromLinkActionCreatorPayload>;
+
+  /**
+   * Updates the standalone search box analytics data to reflect a search submitted by selecting a query suggestion.
+   *
+   * @param payload - The action creator payload.
+   * @returns A dispatchable action.
+   */
+  updateAnalyticsToOmniboxFromLink(
```

Comment:
Feels generally weird to introduce the concept of Omnibox to our Headless user, it might confuse them. From the doc "a search submitted by selecting a query suggestion" could we name it `updateAnalyticsToSuggestionFromLink` so it constrasts with searchFromLink better.

We can keep the "omnibox" term internal only

Reply:
I agree, it feels out of place with the rest of headless, but I went this route to keep the same naming as the `coveo.analytics` package. The state set by this action is used to log this [analytics action](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/features/analytics/analytics-actions.ts#L165) which uses the same wording.

I think it would be more consistent if the same name was used here,
That said, we could say `SearchFromLink` and `OmniboxFromLink` are internal actions that we do not expose?